app/controllers/reports_controller.rb

```diff
@@ -122,6 +136,14 @@ def add_data_to_sheet_for_export_applications(sheet, applications)
     end
   end
 
+  def getCountsForUserSignInReport(one_week_where_clause, two_weeks_where_clause, under_a_month_where_clause, over_a_month_where_clause)
+    @counts = {}
+    @counts["One Week"] = User.where(one_week_where_clause).count
+    @counts["Two Weeks"] = User.where(two_weeks_where_clause).count
+    @counts["Under a month"] = User.where(under_a_month_where_clause).count
+    @counts["Over a month"] = User.where(over_a_month_where_clause).count
+  end
+
   def get_data_for_export_applications
     return UserApplication.joins("inner join Users on users.id = user_applications.user_id")
         .joins("inner join User_informations on user_informations.user_id = users.id")
```

Comment:
@SeanLF The select and joins statement can get a little lost here only because the list of fields is quite large. You might consider moving this list to another method called something like data_for_export_select_fields and so it reduces your method call to

    .joins(...).select(data_for_export_select_fields)
    ...
    ...
    def data_for_export_select_fields
        "user_information.first_name"...
    end

Also, because this is the last statement in the method, you technically don't need a the keyword 'return'